### PR TITLE
Fix: Failed indexes in queries

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

Currently, both `queries` and `orderAttributes` consider index allowed even if it has failed to be created.

With this PR, they will now both properly throw 'Index not found' if index wasn't successfully created.

## Test Plan

- [x] Manual QA on Gitpod (test both queries and orders)

With failed index:

CleanShot 2022-04-05 at 18 39 34
CleanShot 2022-04-05 at 18 45 44

With available index:

CleanShot 2022-04-05 at 18 38 56

CleanShot 2022-04-05 at 18 45 53

## Related PRs and Issues

https://github.com/appwrite/appwrite/issues/2763

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Comment by eldadfux** [comment]
My only question: should this be part of Appwrite or Utopia?

**Comment by Meldiron** [comment]
> My only question: should this be part of Appwrite or Utopia?

I made this mistake 😅 It was part of Utopia PR previously: https://github.com/utopia-php/database/pull/118

Status of the attribute is Appwrite-specific, as worker-database if appwrite-specific. If you use Utopia directly, you don't need to wait for the attribute to get ready - your code gets blocked until it is.

With this reason, I decided to close Utopia PHP PR, and create one for Appwrite.

**Review by TorstenDittmann — APPROVED** [review]

File: app/controllers/api/database.php [hunk 1]
```diff
@@ -31,6 +30,8 @@
 use Appwrite\Network\Validator\IP;
 use Appwrite\Network\Validator\URL;
 use Appwrite\Utopia\Database\Validator\CustomId;
+use Appwrite\Utopia\Database\Validator\Queries as QueriesValidator;
+use Appwrite\Utopia\Database\Validator\OrderAttributes as OrderAttributesValudator;
```
Comment (christyjacob4) [inline app/controllers/api/database.php:34]:
```suggestion
use Appwrite\Utopia\Database\Validator\OrderAttributes as OrderAttributesValidator;
```

File: app/controllers/api/database.php [hunk 2]
```diff
@@ -1742,6 +1743,13 @@ function createAttribute(string $collectionId, Document $attribute, Response $re
             }
         }
 
+        if(!empty($orderAttributes)) {
+            $validator = new OrderAttributesValudator($collection->getAttribute('attributes', []), $collection->getAttribute('indexes', []), true);
```
Comment (christyjacob4) [inline app/controllers/api/database.php:1747]:
```suggestion
            $validator = new OrderAttributesValidator($collection->getAttribute('attributes', []), $collection->getAttribute('indexes', []), true);
```

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Looks good 👍 
Just a typo

**Pull request merged**